cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -116,6 +117,13 @@ public final ClusterStatus deleteClusterWithStatus(ProvisionerContext context, C
         if (jobDetail != null && !jobDetail.getStatus().isTerminated()) {
           return ClusterStatus.RUNNING;
         }
+
+        if (jobDetail != null
+            && jobDetail.getStatus() == RuntimeJobStatus.FAILED
+            && jobDetail.getJobStatusDetail() != null) {
+          LOG.error("Dataproc job failed with the status details: {}",
```

Comment:
Why logging as an error? Typically if we handle and proceed in a normal flow, you don't log it as error (the system expects and handled it).

Reply:
The status details for the failed dataproc job should be displayed as error message in the pipeline logs.

Reply:
Is this the best place to put the LOG.error? Technically deleteClusterWithStatus did not have an error. If the error is during provisioning, look at where getClusterFailureMsg gets called. That maybe a good place.

Reply:
`deleteClusterWithStatus` seems a good place to log the status details. This code path is executed once the pipeline execution has completed (or failed).